test/e2e/network/loadbalancer.go

```diff
@@ -1103,7 +1103,18 @@ var _ = common.SIGDescribe("LoadBalancers ExternalTrafficPolicy: Local", feature
 			framework.Failf("Service HealthCheck NodePort was not allocated")
 		}
 
-		ips := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+		// Only fetch a single v1.NodeInternalIP address from each node.
+		ips := []string{}
```

Comment:
I the assertion we match on the node name, maybe instead of array, ips mapped to their node will be better.
```suggestion
		ips := make(map[string]string)
```

```expectedSuccess := nodes.Items[n].Name == endpointNodeName```

Reply:
On a second thought maybe should assert on all the IPs (internal) of node, as we serve service health server on multiple IPs.
```
ips := make(map[string]set.String)
```
https://github.com/kubernetes/kubernetes/blob/3d6e49e3c24324588c354ff38d5afdb1e27f204f/pkg/proxy/healthcheck/service_health.go#L66